packages/jest-editor-support/src/parsers/babylon_parser.js

```diff
@@ -7,27 +7,31 @@
  * @flow
  */
 
-import {readFileSync} from 'fs';
-
-import {parse as babylonParse} from 'babylon';
+import fs from 'fs';
+import {parse as babelParse} from '@babel/core';
 import {Expect, ItBlock} from './parser_nodes';
+// TODO: Change import to @babel/parser once types exist
 import type {File as BabylonFile} from 'babylon';
 
 export type BabylonParserResult = {
   expects: Array<Expect>,
   itBlocks: Array<ItBlock>,
 };
 
-export const getASTfor = (file: string): BabylonFile => {
-  const data = readFileSync(file).toString();
-  const config = {plugins: ['*'], sourceType: 'module'};
-  return babylonParse(data, config);
+export const getASTfor = (file: string): ?BabylonFile => {
+  const data = fs.readFileSync(file).toString();
+
+  return babelParse(data, {
+    babelrc: true,
```

Comment:
`parse` enables plugins based on the config, so attempting to find the config is required.

Reply:
This isn't possible because we need to differentiate between `flow` and `typescript`. We can't statically support both.

According to @loganfsmyth, we need to use `parse` from `@babel/core` to resolve the plugin list automatically.

Reply:
hmmm... I am just trying to make this backward compatible. 

Yes the `parser` function in `@babel/core` will attempt to read user's config and load the plugins listed there, but 
1. that is not how the original implementation does - it used babylon, which is `@babel/parser` in 7. It passed in the config options instead of depending on user's babel config. 
1. It is true that flow and typescript plugins are incompatible, so how did `plugins:["*"]` ever work? Turns out babel 6 didn't have typescript plugin. So we do not need to include typescript plugin here either. 
1. and yes, with a static list we will lose the ability to include new babel plugins automatically, but I think until babel 7 supports wildcard, this is an acceptable tradeoff. New plugins won't be added frequently, and since we only use this parse function for test files, chances they require these new plugins are even less. And if it does happen, we can simply add the new plugin in the list if needed.

I was able to get vscode-jest working again with the following plugins: 
```
return babelParse(data, {
    plugins: [
      'flow',
      'flowComments',
      'jsx',
      'doExpressions',
      'objectRestSpread',
      'decorators, , {decoratorsBeforeExport: false}',
      'classProperties',
      'asyncGenerators',
      'functionBind',
      'functionSent',
      'dynamicImport',
    ],
    sourceType: 'module',
  });
```
Although more extensive tests are probably needed to ensure full backward compatiblility. Also feel it is a bit slower than babel 6, but I have no historical data to back it up, just want to throw it out there in case it is a common experienced...

Reply:
This Babel 7 PR should definitely be a breaking change. We don't have to worry about backwards compatibility. Also hardcoding the plugin list means TS users can never use Jest.

Reply:
> hardcoding the plugin list means TS users can never use Jest with Babel instead of ts-jest.

1. this parser/package has nothing to do with jest core feature. `jest-editor-support`is mainly to facilitate toolings like vscode-jest to display test blocks in IDE editors, thus the name. 
1. Since this parser clearly doesn't support typescript, so how do we support IDE for typescript today? we used `jest-test-typescript-parser` (again, nothing to do with jest core feature). If you want to take this opportunity to consolidate them that will be fine. A static list doesn't prevent the parser from supporting typescript, you just need to pass a flag then return different list accordingly. 

> This Babel 7 PR should definitely be a breaking change

Just to reiterate, everything we discussed here is under the context of the IDE/tooling support, not jest itself.

Just because we use babel parser to parse user's file doesn't necessarily mean the user is using babel, right? They might use a different mechanism to transpile their code; in such case, a parser relying on user's `.babelrc` will not work, right? What do you envision the breaking change should be for this use case?

Reply:
Sorry, misunderstood, good point. I was thinking in the context of Jest itself, not for this individual package.

In the context of vscode-jest, is there currently a way to pass options to Jest? If so, we can probably support Flow and TS through that in some way. If not, we can go with what you're describing.

Reply:
@connectdotz Updated to the old pattern and enabled most plugins. Also added a TS argument.